docs/search_syntax.md

```diff
@@ -0,0 +1,294 @@
+
+```python
+run.duration == 60
+```
+
+
+Select only the runs where the duration is not 60
+```python
+run.duration != 60
+```
+
+Select only the runs where the duration is greater than 60
+```python
+run.duration > 60
+```
+
+Select only the runs where the duration is greater or equal to 60
+```python
+run.duration >= 60
+```
+
+Select only the runs where the duration is less than 60
+```python
+run.duration < 60
+```
+
+Select only the runs where the duration is less or equal to 60
+```python
+run.duration <= 60
+```
+
+### Example with ```run.archived``` (boolean)
+Select only the runs where the archived attribute is true
+```python
+run.archived == True
+```
+
+Select only the runs where the archived attribute is not true
+```python
+run.archived != True
+```
+
+### Run parameters
+Run parameters can be accessed via parameters.
+![FastTrackML Run List, param filter](images/search_runs_param_filter.png)
+
+### Filtering Runs with Unset Parameters
+
+To filter runs based on whether an attribute is not set, you can use the following syntax:
+
+```python
+run.attribute is None
+```
+This expression will return runs for which the specified attribute is not defined.
+
+Showing all the runs
+![FastTrackML Run List](images/search_runs.png)
+
+Showing only the runs where param1 is not set
+![FastTrackML Run List of not set param](images/search_runs_none_param.png)
+
+### Filter Runs using Regular Expressions
+- ``` re.match() ```
+- ``` re.search() ```
+
+Match finds an exact match at the beginning of a string.
+![FastTrackML Run filter using regular expression match](images/search_runs_regular_expression_match.png)
+
+Search looks for a pattern anywhere in the string.
+![FastTrackML Run filter using regular expression match](images/search_runs_regular_expression_search.png)
+
+### Complex query for run search
+The query selects the runs that meet the following conditions:
+
+
+- run.archived can be either True or False.
+- The duration of run must be greater than 0.
+- The run has to contain a metric named 'TestMetric' and its last recorded value must be greater than 2.5.
+- The name of run should not end with '4'.
+  
+```python
+(run.archived == True or run.archived == False) and run.duration > 0 and run.metrics['TestMetric'].last > 2.5 and not run.name.endswith('4')
+```
+
+## Search metrics examples
+
+### Example with ```metric.name``` (string)
+Select only the metrics where the name exactly matches "TestMetric1"
+```python
+metric.name == "TestRun1"
+```
+
+Select only the metrics where the name is different from "TestMetric1"
+```python
+metric.name != "TestRun1"
+```
+
+Select only the metrics where "Run1" is contained in the run name
+```python
+"Metric1" in metric.name
+```
+
+Select only the metrics where the name starts with "Test"
+```python
+metric.name.startswith('Test')
+```
+
+Select only the metrics where the name ends with "Metric1"
+```python
+metric.name.endswith('Metric1')
+```
```

Comment:
we should list _all_ the string comparison operators here, especially as we gave them for runs too — maybe we just keep `==`, `startswith`, and `re.search`